internal/service/artifacts/api/server.go

```diff
@@ -67,6 +90,55 @@ func (r *ArtifactsServer) GetManagedInfrastructureTemplates(
 func (r *ArtifactsServer) GetManagedInfrastructureTemplate(
 	ctx context.Context, request api.GetManagedInfrastructureTemplateRequestObject) (api.GetManagedInfrastructureTemplateResponseObject, error) {
 
-	// TODO implement me
-	return nil, fmt.Errorf("not implemented")
+	// Get all the ClusterTemplates from the hub cluster.
+	var allClusterTemplates provisioningv1alpha1.ClusterTemplateList
+	err := r.HubClient.List(ctx, &allClusterTemplates)
+	if err != nil {
+		return nil, fmt.Errorf("could not get list of ManagedInfrastructureTemplate across the cluster: %w", err)
+	}
+
+	// Range through the ClusterTemplates and find the one that matches the request's id.
+	// We consider ManagedInfrastructureTemplateId to match ClusterTemplate metadata.name
+	// which is of format ClusterTemplate <spec.name>.<spec.version>.
+	for _, clusterTemplate := range allClusterTemplates.Items {
+		if request.ManagedInfrastructureTemplateId == clusterTemplate.Name {
+			object, err := clusterTemplateToManagedInfrastructureTemplate(clusterTemplate)
+			if err != nil {
+				return nil, err
+			}
+			return api.GetManagedInfrastructureTemplate200JSONResponse(object), nil
+		}
+	}
```

Comment:
Is it possible to use a Get with a selector rather than a List and having to explicitly search through the result set?

Reply:
I've used `client.MatchingFields{"metadata.name": request.ManagedInfrastructureTemplateId}`.
Thanks.